_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,452 @@
+Changes also enhance the `-Wconf` option to allow origin-based filtering `-Wconf:origin=full.path.selector` as in Scala 2. That allows, for example, to exclude certain blessed imports from warnings, or to work around false positives:
+
+```scala
+-Wconf:origin=scala.util.chaining.given:s
+```
+
+These improvements collectively provide developers with more precise and configurable linting tools, enhancing code quality and maintainability.​
+
+### Implicit parameters now warn at call site without `using` keyword [#22441](https://github.com/scala/scala3/pull/22441)
+
+As part of Scala's ongoing migration from the older implicit syntax to the newer, clearer given and using syntax, Scala 3.7 introduces a change regarding method calls involving implicit parameters. Now, explicitly providing arguments to methods defined with implicit parameters without the `using` keyword emits a compiler warning at the call site. This adjustment reduces ambiguity inherent in Scala 2 syntax, where it was unclear whether an explicitly provided argument was intended for an implicit parameter or for the apply method of the resulting object.
+
+```scala
+trait Config
+object Config:
+  def default: Config = ???
+  
+def implicitBased(implicit config: Config) = ???
+def givenBased(using Config) = ???
+
+@main def explicitImplicits = 
+  implicitBased(Config.default)       // warning: Implicit parameters should be provided with a `using` clause.
+  implicitBased(using Config.default) // recommended
+  
+  givenBased(Config.default)       // error: Would not compile without `using``
+  givenBased(using Config.default)
+```
+
+Scala 3.7 provides an automated migration path for existing codebases through the compiler flags -rewrite -source:3.7-migration, which automatically inserts the recommended `using` keywords, streamlining the transition to the new syntax.
+
+### Scala 3 unblocked on Android [#22632](https://github.com/scala/scala3/pull/22632)
+
+Scala 3.7 brings a crucial fix that enhances its compatibility with the Android platform. Previously, developers faced issues when compiling Scala 3 code for Android due to the Android Runtime (ART) enforcing stricter type constraints on lambda expressions compared to the standard Java Virtual Machine (JVM). Specifically, ART requires that the return type of a Single Abstract Method (SAM) interface be a primitive type or explicitly boxed, a condition not mandated by the JVM.​
+
+The update addresses this by modifying the Scala compiler to box the return type of native instantiated methods when the SAM's return type isn't primitive. This change ensures that lambda expressions conform to ART's expectations, preventing runtime errors on Android devices. By aligning the compiler's behaviour with ART's requirements, the Scala 3 development for the Android platform should be unblocked, although it might require recompiling existing libraries using Scala 3.7 or the upcoming Scala 3.3 LTS version containing a backported fix.
```

Comment:
@SethTisue @sjrd I'll be honest that I don't quite understand the underlying issue yet. Can you check if there are not mistakes in this part? 
There were some additional conversations in this area, are there any more conclussions to add here?

Reply:
I wasn't deeply involved with the PR, but AFAICT what is written here is accurate.

Reply:
@ddtthh can you take a look?

Reply:
Maybe this is too detailed for release notes? I think something like this would be enough:

"The bytecode emitted for lambdas involving primitive types was not entirely following the specification. This never caused issues on the JVM, but prevented code from running on Android".